exercises/practice/robot-simulator/tests/robot-simulator.rs

```diff
@@ -1,145 +1,160 @@
+fn changes_east_to_south() {
+    let robot = Robot::new(0, 0, Direction::East);
+    let robot = robot.turn_right();
+    assert_eq!(robot.position(), (0, 0));
+    assert_eq!(robot.direction(), &Direction::South);
 }
 
 #[test]
 #[ignore]
-fn turning_right_from_west_points_the_robot_north() {
-    let robot = Robot::new(0, 0, Direction::West).turn_right();
-    assert_eq!(&Direction::North, robot.direction());
+fn changes_south_to_west() {
+    let robot = Robot::new(0, 0, Direction::South);
+    let robot = robot.turn_right();
+    assert_eq!(robot.position(), (0, 0));
+    assert_eq!(robot.direction(), &Direction::West);
 }
 
 #[test]
 #[ignore]
-fn turning_left_does_not_change_position() {
-    let robot = Robot::new(0, 0, Direction::North).turn_left();
-    assert_eq!((0, 0), robot.position());
+fn changes_west_to_north() {
+    let robot = Robot::new(0, 0, Direction::West);
+    let robot = robot.turn_right();
+    assert_eq!(robot.position(), (0, 0));
+    assert_eq!(robot.direction(), &Direction::North);
 }
 
 #[test]
 #[ignore]
-fn turning_left_from_north_points_the_robot_west() {
-    let robot = Robot::new(0, 0, Direction::North).turn_left();
-    assert_eq!(&Direction::West, robot.direction());
+fn changes_north_to_west() {
+    let robot = Robot::new(0, 0, Direction::North);
+    let robot = robot.turn_left();
+    assert_eq!(robot.position(), (0, 0));
+    assert_eq!(robot.direction(), &Direction::West);
 }
 
 #[test]
 #[ignore]
-fn turning_left_from_west_points_the_robot_south() {
-    let robot = Robot::new(0, 0, Direction::West).turn_left();
-    assert_eq!(&Direction::South, robot.direction());
+fn changes_west_to_south() {
+    let robot = Robot::new(0, 0, Direction::West);
+    let robot = robot.turn_left();
+    assert_eq!(robot.position(), (0, 0));
+    assert_eq!(robot.direction(), &Direction::South);
 }
 
 #[test]
 #[ignore]
-fn turning_left_from_south_points_the_robot_east() {
-    let robot = Robot::new(0, 0, Direction::South).turn_left();
-    assert_eq!(&Direction::East, robot.direction());
+fn changes_south_to_east() {
+    let robot = Robot::new(0, 0, Direction::South);
+    let robot = robot.turn_left();
+    assert_eq!(robot.position(), (0, 0));
+    assert_eq!(robot.direction(), &Direction::East);
 }
 
 #[test]
 #[ignore]
-fn turning_left_from_east_points_the_robot_north() {
-    let robot = Robot::new(0, 0, Direction::East).turn_left();
-    assert_eq!(&Direction::North, robot.direction());
+fn changes_east_to_north() {
+    let robot = Robot::new(0, 0, Direction::East);
+    let robot = robot.turn_left();
+    assert_eq!(robot.position(), (0, 0));
+    assert_eq!(robot.direction(), &Direction::North);
 }
 
 #[test]
 #[ignore]
-fn advance_does_not_change_the_direction() {
-    let robot = Robot::new(0, 0, Direction::North).advance();
-    assert_eq!(&Direction::North, robot.direction());
+fn facing_north_increments_y() {
+    let robot = Robot::new(0, 0, Direction::North);
+    let robot = robot.advance();
```

Comment:
Is this shadowing what you want? It might be slightly confusing to a student?

Reply:
Good point! changed it to `robot_start` and `robot_end`